### ISSUE TITLE
Output logits of GPT-2 not possible. But what about predicted token at each position of context?

### ISSUE THREAD

**Issue by trianxy** [NVIDIA/FasterTransformer#595]
Hi team!

Given an input of shape `[batch_size, sequence_length]`, [this issue](https://github.com/NVIDIA/FasterTransformer/issues/306) mentions that we can't (easily) get the output logits of GPT-2. If we used PyTorch & HuggingFace transformers without FasterTransformer, these would have the shape `[batch_size, sequence_length, vocabulary_size]`.

But what about getting, at least, the predicted token for each token in `sequence_length`, i.e. something of shape `[batch_size, sequence_length]`? 

Note, that when we ask the model to predict the next token of `My name is`, PyTorch does not only predict the token after ` is`, but will also, in the same forward pass, predict the next token after `My` etc. (without looking to the right). This can be helpful if we want to check, in **one** forward pass, whether the continuation would have been `My name is FIRSTNAME LASTNAME`.
